microsoft/testsuites/dpdk/dpdkovs.py

```diff
@@ -0,0 +1,275 @@
+        # _install just checks out our resources and sets up the version info
+        # TODO: Add option to select which OVS version to use other than latest
+
+        self._install_os_packages()
+
+        # checkout git and get latest version tag
+        git = self.node.tools[Git]
+        self.repo_dir = git.clone(
+            "https://github.com/openvswitch/ovs.git",
+            cwd=self.node.working_path,
+        )
+        latest_version_tag = git.get_tag(cwd=self.repo_dir)
+        git.checkout(latest_version_tag, cwd=self.repo_dir)
+
+        # parse version info from git tag to validate the right dpdk version is used
+        match = self._version_regex.search(latest_version_tag)
+        if not match or not all(
+            [match.group("major"), match.group("minor"), match.group("patch")]
+        ):
+            fail(
+                f"Could not match version tag '{latest_version_tag}' "
+                f"with regex '{self._version_regex.pattern}'"
+            )
+        else:
+            major, minor, patch = map(
+                int, [match.group("major"), match.group("minor"), match.group("patch")]
+            )
+            self.ovs_version = VersionInfo(major, minor, patch)
+
+        return True
+
+    def _check_ovs_dpdk_compatibility(self, dpdk_tool: DpdkTestpmd) -> None:
+        dpdk_version = dpdk_tool.get_dpdk_branch()
+        # confirm supported ovs:dpdk version pairing based on
+        # https://docs.openvswitch.org/en/latest/faq/releases/
+        # to account for minor releases check release is below a major version threshold
+        ovs_dpdk_version_pairings = [
+            ("2.13.0", 18),  # if below 2.13.0 then dpdk version is 18
+            ("2.15.0", 19),  # and so on
+            ("2.17.0", 20),
+            ("2.18.0", 21),
+        ]
+        # check is version too low
+        if self.ovs_version < "2.11.0":
+            fail(
+                "OVS < 2.10 is not supported with any supported "
+                "DPDK version as of 2/8/2022"
+            )
+        # check if version too high (pairing is unknown)
+        elif self.ovs_version > "2.18.0":
+            raise NotImplementedError(
+                (
+                    f"Not implemented OVS version {self.ovs_version}. "
+                    "Please update the version requirement table in "
+                    "dpdkovs.py:check_ovs_dpdk_compatibilty with new data"
+                    "from https://docs.openvswitch.org/en/latest/faq/releases/"
+                )
+            )
+        else:
+            # iterate confirmed supported versions
+            for version_limit in ovs_dpdk_version_pairings:
+                ovs_upper_limit, dpdk_version_requirement = version_limit
+                if self.ovs_version < ovs_upper_limit:
+                    # located the correct range, check if versions match
+                    if dpdk_version.major == dpdk_version_requirement:
+                        break
+                    else:
```

Comment:
This level of `else` is not necessary. If the `if` meets, it breaks out, if not, it's always failed.